src/test/java/com/ge/predix/uaa/token/lib/HttpServletRequestUtilTest.java

```diff
@@ -143,4 +153,18 @@ public void testGetZoneNameWithHeaderAndBaseDomains(final String requestHostname
                         null } };
     }
 
+    @DataProvider(name = "subdomainDataProvider")
+    private Object[][] subdomainDataProvider() {
+        return new Object[][] {
+                { "zone1.acs.com", Arrays.asList("acs.com", "guardians.com"),
+                        Collections.singletonList("Predix-Zone-Id"), null, false },
+                { "acs.com", Arrays.asList("acs.com", "guardians.com"),
+                            Collections.singletonList("Predix-Zone-Id"), null, false },
+                { "zone1.acs.com", Arrays.asList("acs.com", "guardians.com"),
+                        Collections.singletonList("Predix-Zone-Id"), "zone1", true },
+                { "acs.com", Arrays.asList("acs.com", "guardians.com"),
+                            Collections.singletonList("Predix-Zone-Id"), null, true }};
+
+    }
+
 }
```

Comment:
@FrankGasparovic, 
Missing test use cases:
- enableZonesAsSubdomains=false and zone is passed in a header and as a subdomain.  Should return zone from the header
- enableZonesAsSubdomains = false and zone is passed in a header.  Should return zone from the header
- enableZonesAsSubdomains = true and zone is passed in a header.  Should return zone from the header